pennylane/taskify.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains functionality to enable task-based workflows with devices.
+"""
+from typing import Union
+
+import pennylane as qml
+from pennylane.devices import TaskQubit
+
+import dask.distributed as dist
```

Comment:
since this imports TaskQubit and works specifically with it, this feels like it should be part of the `task_qubit.py` module (or maybe a subpackage) 🤔

Reply:
I think something like this would be well suited to an executor-like module (again, decoupling problem definition from where it runs). Happy to move this into the `task_qubit.py` file for now if you think it makes sense.

Reply:
I'm not 100% sure the best organization, but yeah it does feel a little odd (at this stage) to have a top-level module that is connected to the implementation details of a optional dependency device 🤔